src/test/java/org/jvnet/hudson/main/JenkinsRuleTimeoutTest.java

```diff
@@ -156,9 +156,10 @@ public void run() {
         }
     }
 
-    @Test @WithTimeout(20)
+    @Test 
+    @WithTimeout(15)
     public void withTimeoutPropagation() throws Exception {
-        Thread.sleep(1000 * 30);
+        Thread.sleep(1000 * 15);
```

Comment:
Not following how this could be anything but flaky. You are setting a 15s timeout on the test and then sleeping for 15s, so the test will sometimes get the timeout (and pass, due to `thrown.expect(TestTimedOutException.class)`?) and sometimes not get the timeout and so fail with the assertion?

Reply:
My concern with this test is that if you remove the annotation it still passes and so, does not test what it's expected to test.

I was too optimistic about the guarantee from sleep, moving to 20s.